### PR TITLE
Adding test for score package

### PR DESCRIPTION
**Pull request by devtomnl**
* Adding test for Score package

It includes: 
New Helper struct and file that mocks located in score/testutil:
1. GetLastCommittee
2. GetLastCertificate
3. GetMempoolTxsBySize
4. Response from Gossip that checks if we got a topics.Score propagated correctly 

Next steps:
Add negative paths such as, failing to respond with GetLastCommittee, GetLastCertificate, GetMempoolTxsBySize

### CODE REVIEW

File: pkg/core/data/transactions/distribute.go [hunk 1]
```diff
@@ -169,8 +169,9 @@ func MarshalDistribute(r *bytes.Buffer, s DistributeTransaction) error {
 		}
 	}
 
-	if len(s.BgPk.BG.Y) == 0 || len(s.BgPk.AG.Y) == 0 {
-		return errors.New("could not find a valid pub key reward on the DistributeTransaction")
+	//TODO: check if panic is the most correct way of doing this
+	if s.BgPk == nil || (s.BgPk.BG == nil || s.BgPk.AG == nil) || len(s.BgPk.BG.Y) == 0 || len(s.BgPk.AG.Y) == 0 {
+		panic("could not find a valid pub key reward on the DistributeTransaction")
 	}
```
Comment (autholykos) [inline pkg/core/data/transactions/distribute.go]: I am fine keeping the panic here

File: pkg/core/consensus/score/testutil.go [hunk 2]
```diff
@@ -18,9 +18,7 @@ import (
 // Helper for reducing test boilerplate
 type Helper struct {
 	*consensus.Emitter
-	lock               sync.RWMutex
-	failOnFetching     bool
-	failOnVerification bool
```
Comment (autholykos) [inline pkg/core/consensus/score/testutil.go]: please rebase this commit with the previous one

File: pkg/core/consensus/score/testutil.go [hunk 3]
```diff
@@ -11,15 +13,11 @@ import (
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/rpcbus"
 	log "github.com/sirupsen/logrus"
-	"sync"
-	"time"
 )
```
Comment (autholykos) [inline pkg/core/consensus/score/testutil.go]: Please rebase

**Review by autholykos — CHANGES_REQUESTED** [review]

File: pkg/core/data/transactions/distribute.go [hunk 4]
```diff
@@ -146,6 +147,9 @@ func UDistribute(r *rusk.DistributeTransaction, t *DistributeTransaction) error
 
 //MarshalDistribute into a buffer
 func MarshalDistribute(r *bytes.Buffer, s DistributeTransaction) error {
+	if s.Tx == nil || len(s.Tx.Outputs) == 0 || s.Tx.Outputs[0].Note == nil {
+		return errors.New("could not find the coinbase reward on the DistributeTransaction")
+	}
```
[the remark this answers is not in the document]
Comment (devtomnl) [reply pkg/core/data/transactions/distribute.go]: great, fixed.

File: pkg/core/data/transactions/distribute.go [hunk 5]
```diff
@@ -165,6 +169,10 @@ func MarshalDistribute(r *bytes.Buffer, s DistributeTransaction) error {
 		}
 	}
 
+	if len(s.BgPk.BG.Y) == 0 || len(s.BgPk.AG.Y) == 0 {
+		return errors.New("could not find a valid pub key reward on the DistributeTransaction")
+	}
```
[the remark this answers is not in the document]
Comment (devtomnl) [reply pkg/core/data/transactions/distribute.go]: great, fixed.

File: pkg/core/data/transactions/distribute.go (hunk 1 above)
Comment (devtomnl) [reply pkg/core/data/transactions/distribute.go]: ok

File: pkg/core/consensus/score/testutil.go (hunk 2 above)
Comment (devtomnl) [reply pkg/core/consensus/score/testutil.go]: done

File: pkg/core/consensus/score/testutil.go (hunk 3 above)
Comment (devtomnl) [reply pkg/core/consensus/score/testutil.go]: done

**Pull request merged**